state/machine.go

```diff
@@ -148,6 +148,10 @@ type machineDoc struct {
 
 	// Hostname records the machine's hostname as reported by the machine agent.
 	Hostname string `bson:"hostname,omitempty"`
+
+	// DisplayName records the machine's display-name (currently
+	// only OCI and MAAS providers use this)
+	DisplayName string `bson:"display-name,omitempty"`
```

Comment:
We don't want to add this field to the machine doc. It needs to go from/to the `instanceData` doc.

Reply:
Changed to use the `CloudInstance` and `instanceData` :+1: